generator/src/main/scala/Prog.scala

```diff
@@ -343,7 +344,24 @@ class Prog(memsize: Int, veccfg: Map[String,String], run_twice: Boolean)
       while (!is_seqs_active_empty)
       {
         val seq = rand_pick(seqs_active)
-        add_inst(seq.next_inst())
+	if(loop) {
+	  val inst = seq.next_inst()
+	  val branch_filter = (x: Operand) =>
+      	    x.isInstanceOf[Label] && x.asInstanceOf[Label].label.indexOf("branch_patch") != -1
+    	  val branch_patch = inst.operands.indexWhere(branch_filter)
+	  val jalr_filter1 = (x: Operand) =>
+      	    x.isInstanceOf[Label] && x.asInstanceOf[Label].label.indexOf("jalr_patch1") != -1
+    	  val jalr_patch1 = inst.operands.indexWhere(jalr_filter1)
+	  val jalr_filter2 = (x: Operand) =>
+      	    x.isInstanceOf[Label] && x.asInstanceOf[Label].label.indexOf("jalr_patch2") != -1
+    	  val jalr_patch2 = inst.operands.indexWhere(jalr_filter2)
+    	  if (jalr_patch1 == -1 && branch_patch == -1 && jalr_patch2 == -1){
```

Comment:
What is going on here? What are these patch labels?

I also really don't like how you did the filtering, there has to be a better way to do that.